ui/imports/shared/views/AssetsViewNew.qml

```diff
@@ -0,0 +1,333 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import QtQml.Models 2.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Popups.Dialog 0.1
+
+import AppLayouts.Wallet.controls 1.0
+import shared.controls 1.0
+import shared.popups 1.0
+import utils 1.0
+
+import SortFilterProxyModel 0.2
+
+
+Control {
+    id: root
+
+    /**
+      Expected model structure:
+
+        key         [string] - unique identifier of a token, e.g "0x3234235"
+        symbol      [string] - token's symbol e.g. "ETH" or "SNT"
+        name        [string] - token's name e.g. "Ether" or "Dai"
+        icon        [url]    - token's icon url
+        balance     [double] - tokens balance is the commonly used unit, e.g. 1.2 for 1.2 ETH, used
+                               for sorting and computing market value
+        balanceText [string] - formatted and localized balance. This is not done internally because
+                               it may depend on many external factors
+        error       [string] - error message related to balance
+
+        marketDetailsAvailable [bool]   - specifies if market datails are available for given token
+        marketDetailsLoading   [bool]   - specifies if market datails are available for given token
+        marketPrice            [double] - specifies market price in currently used currency
+        marketChangePct24hour  [double] - percentage price change in last 24 hours, e.g. 0.5 for 0.5% of price change
+
+        communityId   [string] - for community assets, unique identifier of a community, e.g. "0x6734235"
+        communityName [string] - for community assets, name of a community e.g. "Crypto Kitties"
+        communityIcon [url]    - for community assets, community's icon url
+
+        position    [int]  - if custom order available, display position defined by the user via token management
+        canBeHidden [bool] - specifies if given token can be hidden (e.g. ETH should be always visible)
+    **/
+    property var model
```

Comment:
The component can be instantiated and be in valid state without model, which can be set later. I added small fix internally to avoid warning when not defined.